### PR TITLE
[EasyErrorHandler] Fix ApiPlatformValidationErrorResponseBuilder for API Platform 2.7

### PR DESCRIPTION
**Pull request by alexndlm**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Comment by alexndlm** [comment]
PhpStan errors were being ignored in this PR - https://github.com/eonx-com/easy-monorepo/pull/1039

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php [hunk 1]
```diff
@@ -32,11 +34,19 @@ public static function supports(Throwable $throwable): bool
     {
         $message = $throwable->getMessage();
 
+        // TODO: refactor in 5.0. Use the ApiPlatform\Symfony\Bundle\ApiPlatformBundle class only.
+        $invalidArgumentExceptionClass = stdClass::class;
```
Comment (akl-95) [inline packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:38]: Can we set `null` instead?

Comment (akl-95) [inline packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:38]: Can we set it to `null` instead?

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php [hunk 2]
```diff
@@ -32,11 +34,19 @@ public static function supports(Throwable $throwable): bool
...
+        }
+
         return match ($throwable::class) {
-            InvalidArgumentException::class => \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $message) === 1,
+            $invalidArgumentExceptionClass, LegacyInvalidArgumentException::class =>
+                \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $message) === 1,
             MissingConstructorArgumentsException::class =>
                 \preg_match(self::MESSAGE_PATTERN_NO_PARAMETER, $message) === 1,
             UnexpectedValueException::class =>
+                \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $message) === 1 ||
```
Comment (akl-95) [inline packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:49]: What is the reason to add it in that case?

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php [hunk 3]
```diff
@@ -85,6 +105,20 @@ public function buildData(Throwable $throwable, array $data): array
             $matches = [];
 
             switch (1) {
+                case \preg_match(self::MESSAGE_PATTERN_TYPE_ERROR, $throwable->getMessage(), $matches) === 1:
+                    $data['violations'] = [
+                        $matches[1] => [
+                            $matches[3] === 'NULL'
+                                ? (new NotNull())->message
+                                : \sprintf(
```
Comment (akl-95) [inline packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:113]: Can we write it inline to be consistent with line 90?

**Review by akl-95 — CHANGES_REQUESTED** [review]

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 2 above)
Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:49]: I didn't investigate deeply. It looks like the API platform started returning the Symfony Serializer exception during deserialization.

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 3 above)
Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php]: Refactored in some another way.

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 1 above)
Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php]: Yes.

Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php]: Refactored.

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 2 above)
Comment (akl-95) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:50]: We should exactly understand what we do. So please investigate why this error is possible here and why we didn't add it initially.

Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:50]: Before 2.7 API Platform throw `InvalidArgumentException` in `\ApiPlatform\Core\Serializer\AbstractItemNormalizer::validateType`

Now it throws `UnexpectedValueException` in `\ApiPlatform\Serializer\AbstractItemNormalizer::validateType`

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 1 above)
Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php]: During updating the API platform on Payments, I have an ambitious plan to make the API platform return a validation error. Then we can normally rethink this and similar classes.

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 2 above)
Comment (akl-95) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:50]: Well, does it then mean we don't need to check `$invalidArgumentExceptionClass` above and we need only `LegacyInvalidArgumentException` there?

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 1 above)
Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php]: We need both, in version `2.7` `InvalidArgumentException` at least uses in  `\ApiPlatform\Serializer\AbstractItemNormalizer::denormalizeCollection`.

File: packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php (hunk 2 above)
Comment (alexndlm) [reply packages/EasyErrorHandler/src/Bridge/Symfony/Builder/ApiPlatformValidationErrorResponseBuilder.php:50]: We need both, in version `2.7` `InvalidArgumentException` at least uses in `\ApiPlatform\Serializer\AbstractItemNormalizer::denormalizeCollection`.

**Review by akl-95 — APPROVED** [review]

**Review by itorgov — APPROVED** [review]

**Review by roman-eonx — APPROVED** [review]

**Review by natepage — APPROVED** [review]

**Pull request merged**